adhocteam/Head-Start-TTADP
# Prevent updating of reports

**Description of change**

Reports are now placed in a read only mode if
 * The report's status is `submitted` or `approved`
 * The user viewing the report is NOT the author or a collaborator

In read only mode the user can only view the review page.

Also added a button/API call to move a report from `submitted` back to `draft` mode, allowing the author or collaborators to make edits after submitting but before approval.

**How to test**
1) Pull down
2) Fill out and submit an activity report
3) Note after submitting you see the `submitted` review page and you cannot browse to other pages of the report
4) Bring the report back to `draft` status
5) Submit the report again
6) As the approving manager pull up the report, note you are forced to stay on the review page, set the status to `Needs Action`
7) As the author pull up the report, note you can edit the report as if it had a status of `draft`
8) Resubmit the report and approve with the approving manager
9) Note the report cannot be edited once approved

Keeping in draft until a few of the other report PRs get merged. Still need to update API docs with the new route.

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/285

**Checklist**

    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
